src/euphorie/client/browser/training.py

```diff
@@ -276,17 +294,19 @@ def question_intro_url(self):
         if not getattr(survey, "enable_web_training", False):
             return ""
         view_name = "slide_question_success"
-        if survey.listFolderContents(
-            {"portal_type": "euphorie.training_question"}
+        if api.content.find(
+            obj=survey,
+            portal_type="euphorie.training_question",
         ) and self.training_status not in ("correct", "success"):
             view_name = "slide_question_intro"
         return "{}/@@{}".format(self.context.absolute_url(), view_name)
 
     @property
     @view_memoize
     def questions(self):
-        survey = self.webhelpers._survey
-        return survey.listFolderContents({"portal_type": "euphorie.training_question"})
+        training = self.get_or_create_training()
+        answer_history = loads(training.answers)
+        return list(answer_history)
```

Comment:
This is changed to return the question ids, right?
Then this bubbles up into a lot of changes in the code that otherwise could be avoided.

What about having a `question_ids` property that returns the question ids and restore the return type of the `questions` property?
Would this help in having a smaller patch?

Reply:
But the point was to change the parts in the code that were getting the objects without needing them. Most calls of `questions()` actually don't need the objects. I can of course change it back but then your concern is not addressed.

Reply:
I expressed myself poorly then.

What I am trying to achieve is to have both the properties `questions` and `question_ids`, in that way:
- `self.question_ids` will be used the many times you just need the object ids
- `self.questions` will be used the few time you need the objects and it will be

Does it make more sense now?

Reply:
Yes, but why? There is one or maybe two cases where we need the objects. Why do we need a method for that?

Reply:
In other words, doing it like that will not make the patch smaller.

Reply:
There are many reasons and I would not like to waste lot of time with that.

The premises:
1. This PR already renames:
   - `previous_question` -> `previous_question_id`
   - `next_question` -> `next_question_id`
2. the code already has a `question_id` attribute and a `question` property.

So "the main because" is that as a casual "code reader/debugger" I would not expect `self.questions` to return ids, but rather objects.

This reason alone might already justify to rename the method `questions` to `question_ids`.
I think you agree with me that it would increase the consistency of the code, which is a desirable thing.
Keeping a `questions` method that returns the object might be handy but I am not sure it is needed.

Also I do not know is all this changes will break some other projects that depend on this package.

It might also be a good thing to start adding some deprecation warnings instead of just removing renaming methods.

Reply:
I'm happy to rename `questions` to `question_ids`. The training feature is completely new and not even merged yet, so there can't be any code that depends on this. Therefore, and because you originally remarked that we should not get the objects if we don't need them, I wouldn't bother keeping a method that gets the objects.